src/Microsoft.AspNetCore.WebUtilities/HttpRequestStreamReader.cs

```diff
@@ -132,9 +110,9 @@ protected override void Dispose(bool disposing)
 
         public override int Peek()
         {
-            if (_stream == null)
+            if (_disposed)
             {
-                throw new ObjectDisposedException("stream");
+                throw new ObjectDisposedException(nameof(HttpRequestStreamReader));
```

Comment:
I wouldn't pass anything into the ctor here. The type name is not the "object name" - it's pretty useless here and generates noise.